### PR TITLE
irmin-pack: Diffs for gc benchmark

### PR DESCRIPTION
**Pull request by Ngoguey42**

### CODE REVIEW

**Review by icristescu — APPROVED** [review]

File: CHANGES.md [hunk 1]
```diff
@@ -20,12 +20,14 @@
...
     demonstration of how it works with the new `split` function. (#2126,
     @metanivek)
+  - Add GC generation in GC stats (#2129, @Ngoguey42)
 
 ### Fixed
 
 - **irmin-pack**
   - Fix read-only opening flags of mapping for read-only opening of stores that
     do not have read-write rights on the file system (#2121, @Ngoguey42)
+  - Correctly report stats at the end of GC (#2129, @Ngoguey42)
```
Comment (metanivek) [inline CHANGES.md:30]: Do we need changelog entries since the new stats haven't been released yet?

File: src/irmin-pack/unix/ext.ml [hunk 2]
```diff
@@ -284,7 +284,7 @@ module Maker (Config : Conf.S) = struct
           let start_exn ?(unlink = true) ~use_auto_finalisation t commit_key =
             match t.running_gc with
             | Some _ ->
-                [%log.info "Repo is alreadying running GC. Skipping."];
```
Comment (metanivek) [inline src/irmin-pack/unix/ext.ml:287]: What was the reason for making this debug? I could see this log being helpful in octez logs if somehow the GC gets stuck. It also corresponds with the `log.info` for starting a GC.

File: CHANGES.md (hunk 1 above)
Comment (Ngoguey42) [reply CHANGES.md:30]: Good point. That's not needed

File: src/irmin-pack/unix/ext.ml (hunk 2 above)
Comment (Ngoguey42) [reply src/irmin-pack/unix/ext.ml:287]: This was flooding my replays at some point but I think it can be added back

Comment (metanivek) [reply src/irmin-pack/unix/ext.ml:287]: Are you planning to change this back to info?

**Pull request merged**